cpp/src/phonenumbers/phonenumberutil.cc

```diff
@@ -753,20 +762,23 @@ PhoneNumberUtil::PhoneNumberUtil()
       country_code_to_non_geographical_metadata_map_(
           new std::map<int, PhoneMetadata>) {
   Logger::set_logger_impl(logger_.get());
-  // TODO: Update the java version to put the contents of the init
-  // method inside the constructor as well to keep both in sync.
-  PhoneMetadataCollection metadata_collection;
-  if (!LoadCompiledInMetadata(&metadata_collection)) {
-    LOG(DFATAL) << "Could not parse compiled-in metadata.";
-    return;
-  }
+}
+
+PhoneNumberUtil::~PhoneNumberUtil() {
+  gtl::STLDeleteContainerPairSecondPointers(
+      country_calling_code_to_region_code_map_->begin(),
+      country_calling_code_to_region_code_map_->end());
+}
+
+// Initializes PhoneNumberUtil fields by using data from PhoneMetadataCollection.
+void PhoneNumberUtil::InitializeFromMetadata(PhoneMetadataCollection& metadata_collection){
   // Storing data in a temporary map to make it easier to find other regions
   // that share a country calling code when inserting data.
   std::map<int, std::list<string>* > country_calling_code_to_region_map;
   for (RepeatedPtrField<PhoneMetadata>::const_iterator it =
-           metadata_collection.metadata().begin();
-       it != metadata_collection.metadata().end();
-       ++it) {
+          metadata_collection.metadata().begin();
```

Comment:
nit: spaces here don't align with previous code - I'm not totally sure which one is correct.